### PR TITLE
Connect new search functionality and endpoints to FE #106

### LINKED ISSUE #106: Search FE MT: Test that search is connected on the FE
Now that the search team has finished their work, we must conduct testing to connect to the new search endpoint.

The new endpoint features an enhanced search query. We'll need to test this improvement and update our code accordingly to reflect it.

**Acceptance Criteria:**

- [x]  Modify the code to grab the search query
- [x]  Conduct testing on the search query
- [x]  Update the constant.js to reflect the new endpoint

**Content AC**

- [x] Metadata to be the same as VT https://github.com/aemsites/volvotrucks-na/issues/27

### PR DESCRIPTION
**Pull request by santi-homps**
SUGGESTED TESTING PATH:
Go to main page **->** click on the search icon 🔍 **->** write something **->** check autosuggest **->** press enter 
 **->** check results **->** interact with filters **->**  check filters

Fix #106

Test URLs:
- Before: https://main--vg-macktrucks-com--volvogroup.aem.page/
- After: https://106-fe-search-connect--vg-macktrucks-com--volvogroup.aem.page/

Other markets:

Canada:

- Before: https://main--macktrucks-ca--volvogroup.aem.page/
- After: https://106-fe-search-connect--macktrucks-ca--volvogroup.aem.page/

### CODE REVIEW

**Review by jonatan-lledo-netcentric — APPROVED** [review]

File: blocks/header/header.js [hunk 1]
```diff
@@ -124,16 +126,14 @@ const mobileActions = () => {
...
   const actions = document.createRange().createContextualFragment(`
-    ${SEARCH_DISABLED.toLowerCase() === 'false' ? `
-    <a
-      href="/search"
-      aria-label="${searchLabel}"
-      class="${blockClass}__search-button ${blockClass}__action-link ${blockClass}__link"
-    >
-      <span class="icon icon-search" aria-hidden="true"></span>
-    </a>` : ''}
+    ${SEARCH_DISABLED.toLowerCase() === 'true' ? '' : searchEl}
```
Comment (alexiscoelho) [inline blocks/header/header.js:136]: {SEARCH_DISABLED?.toLowerCase()

File: blocks/search/autosuggest.js [hunk 2]
```diff
@@ -1,25 +1,28 @@
-import { createElement } from '../../scripts/common.js';
-import { autosuggestQuery, fetchData } from './search-api.js';
+import { createElement, getLocale } from '../../scripts/common.js';
+import { autosuggestQuery, fetchData, TENANT } from '../../scripts/search-api.js';
 
 const autoSuggestClass = 'autosuggest-results-item-highlighted';
 
 export function fetchAutosuggest(term, autosuggestEle, rowEle, func) {
   const fragmentRange = document.createRange();
+  const language = getLocale();
```
Comment (alexiscoelho) [inline blocks/search/autosuggest.js:8]: I think you should rename these variables to:
`const locale = getLocale();`
`const language = language.split('-')[0].toUpperCase();`

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: blocks/header/header.js (hunk 1 above)
Comment (santi-homps) [reply blocks/header/header.js]: DONE

File: blocks/search/autosuggest.js (hunk 2 above)
Comment (santi-homps) [reply blocks/search/autosuggest.js]: DONE

**Review by amglebihan — APPROVED** [review]

**Comment by jonatan-lledo-netcentric** [comment]
please, fix the conflicts before QA testing

**Pull request merged**